File: Programming EF Code First/chapter6/Chapter6/Chapter 6 VB/DataAnnotations/DataAccess/BreakAwayContext.vb
Imports System.Data.Entity
Imports Model
Imports System.Data.Common

Namespace DataAccess
  Public Class BreakAwayContext
	  Inherits DbContext
	Public Sub New()
	End Sub

	Public Sub New(ByVal databaseName As String)
		MyBase.New(databaseName)
	End Sub

	Public Sub New(ByVal connection As DbConnection)
		MyBase.New(connection, contextOwnsConnection:= False)
	End Sub

	Public Property Destinations() As DbSet(Of Destination)
	Public Property Lodgings() As DbSet(Of Lodging)
	Public Property Trips() As DbSet(Of Trip)
	Public Property People() As DbSet(Of Person)
  End Class
End Namespace

File: Programming EF Code First/chapter6/Chapter6/BAGA_DataAnnotations/DataAccess/BreakAwayContext.cs
using System.Data.Entity;
using Model;
using System.Data.Common;

namespace DataAccess
{
  public class BreakAwayContext : DbContext
  {
    public BreakAwayContext()
    { }

    public BreakAwayContext(string databaseName)
      : base(databaseName)
    { }

    public BreakAwayContext(DbConnection connection)
      : base(connection, contextOwnsConnection: false)
    { }

    public DbSet<Destination> Destinations { get; set; }
    public DbSet<Lodging> Lodgings { get; set; }
    public DbSet<Trip> Trips { get; set; }
    public DbSet<Person> People { get; set; }
  }
}